.github/workflows/maven-build.yml

```diff
@@ -0,0 +1,102 @@
+name: build on maven
+
+on:
+  watch:
+    types: [started]
+  pull_request:
+    types: [opened, reopened, edited, synchronize, ready_for_review]
+  push:
+    branches:
+      - main
+      - master
+      - '**'
+  workflow_dispatch:
+
+jobs:
+  verify:
+    name: verify with maven
+    runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: "-Xmx4096m -Xms2048m -XX:MaxMetaspaceSize=4096m -Xss8m"
+      SKIP_NPM_CONFIG: false
+      NPMREGISTRY: https://registry.npmjs.org
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up JDK 8 & 11 for x64
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        architecture: x64
+        java-version: |
+          8
+          11
+
+    - uses: s4u/maven-settings-action@v2.8.0
+      with:
+        sonatypeSnapshots: true
+
+    - name: Build the Maven verify phase
+      run: mvn -B -V clean verify -Prun-its -Pci --global-toolchains .github/workflows/toolchains.xml
+  
+  build:
+    name: Build and push
+    runs-on: ubuntu-latest
+    needs: verify
+    if: ${{ github.event_name == 'push' }}
+    env:
+      MAVEN_OPTS: "-Xmx4096m -Xms2048m -XX:MaxMetaspaceSize=4096m -Xss8m"
+      SKIP_NPM_CONFIG: false
+      NPMREGISTRY: https://registry.npmjs.org
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up JDK 8 & 11 for x64
+      uses: actions/setup-java@v3
+      with:
+        distribution: 'temurin'
+        architecture: x64
+        java-version: |
+          8
+          11
+
+    - uses: s4u/maven-settings-action@v2.8.0
+      with:
+        servers: |
+          [{
+              "id": "sonatype-nexus-snapshots",
+              "username": "${{ secrets.SONATYPE_BOT_USERNAME }}",
+              "password": "${{ secrets.SONATYPE_BOT_TOKEN }}"
+          }]
+
+    - name: Build with toolchains
+      run: mvn -B -V clean install -DskipNpmConfig=false --global-toolchains .github/workflows/toolchains.xml
+
+    - name: Deploy the artifact
+      run: mvn help:effective-settings -B -V -DskipTests=true -DskipNpmConfig=false deploy -e --global-toolchains .github/workflows/toolchains.xml
+    
+    - name: Checkout tools repo
+      uses: actions/checkout@v4
+      with:
+        repository: Commonjava/commonjava-images
+        path: commonjava-images
+    
+    - name: Login into Quay
+      uses: docker/login-action@v3
+      with:
+        registry: quay.io
+        username: ${{ secrets.QUAY_USERNAME }}
+        password: ${{ secrets.QUAY_TOKEN }}
+    
+    - name: Build and push
+      uses: docker/build-push-action@v5
+      with:
+        context: ./commonjava-images/indy/
+        file: ./commonjava-images/indy/multistage.Dockerfile
+        platforms: linux/amd64
+        push: true
+        tags: |
+          quay.io/factory2/indy:latest
```

Comment:
A problem here is the tag. Sometimes we will push to branch tag like 3.3.x, 3.2.x(may be obsolete future, but it is still used now IIRC). Is it possible to do?

Reply:
it is possible, but won't that be easier to build in pipeline instead of here?

Reply:
Actually the current branch tag is working as the "latest" tag for each branch to do the merge build. For example, all images on branch 3.3.x for code merge build are automatically pushed to "3.3.x" tag now. 
I think the pipeline is still working for manual build but not automatically merge build, please correct me if I'm wrong.